dask_expr/_shuffle.py

```diff
@@ -740,6 +740,29 @@ def _lower(self):
         return SetPartition(self.frame, self._other, self.drop, divisions)
 
     def _simplify_up(self, parent):
+        from dask_expr._expr import Head, Tail
+
+        # TODO, handle setting index with other frame
```

Comment:
Injecting an Assign is probably the easiest way of getting there

Reply:
Agreed.  I started doing this but it got complicated, so I stopped.